GSA/searchgov-spider
# SRCH-5751 Low Output Fixes

## Summary
Story: https://cm-jira.usa.gov/browse/SRCH-5751
More details about low output [here](https://docs.google.com/spreadsheets/d/1PPx78a10I22EqeXNztqVIgzgK8T9WENcHhvlzYfNrSQ/edit?gid=918090254#gid=918090254) with the domains im concentrating on at the top, high-lit and with "before" and "after results"
- Changes include: 
  - Add "allow_query_string" spider argument to support sites (one site - https://www.herc.research.va.gov/) whose pages all end in some variation of `page.asp?id=<page_name>`.  For all other sites, URLs with query strings continue to be ignored.
  - Adjusted `FILTER_EXTENSIONS` to:
    - Allow coldfusion sites (`.cfm`)
    - No longer allow a few other extensions I noticed in errors messages that are not text but were being downloaded to check for content-type (`.hdf`, `.nc4`, `.geojson` )
  - Added `va-link` to list of tags that we look in for links - to support sites (one site - https://www.va.gov/resources/) that use this custom tag.
  - Removed a few sites from crawl-sites.json at Jim's direction.
  - Fixing issue where a change to the LOG_LEVEL setting in `search_gov_crawler/search_gov_spiders/settings.py` does not change logging level of json logs.
    - Added an optional env var `SCRAPY_LOG_LEVEL` to allow controlling this.  Defaults to "INFO".
  - Various small formatting and linting changes

### Testing
- Run `scrapy crawl` for a few different cases (No need to run any or all of these to completion as some may take a while) 
  - Various test sites to ensure basic functionality is unchanged 
  - Each of the low-output sites that should now produce links.  Confirm that they do produce links:
    - `domain_spider -a start_urls="https://www.research.va.gov/" -a allowed_domains="research.va.gov"`
    - `domain_spider -a start_urls="https://www.queri.research.va.gov/" -a allowed_domains="queri.research.va.gov"`
    - `domain_spider -a start_urls="https://www.cider.research.va.gov" -a allowed_domains="cider.research.va.gov"`
    - `domain_spider -a start_urls="https://www.va.gov/resources/" -a allowed_domains="va.gov/resources/"`
    - `domain_spider -a start_urls="https://www.herc.research.va.gov/" -a allowed_domains="herc.research.va.gov" -a allow_query_string=True`
- Ensure benchmark still works, e.g.:
  - `python benchmark.py -d "quotes.toscrape.com" -u "https://quotes.toscrape.com"`
  - `python benchmark.py -d "herc.research.va.gov" -u "https://www.herc.research.va.gov/" -qs`
  - `python benchmark.py -f ../tests/search_gov_spiders/crawl-sites-test.json`

### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.

#### Functionality Checks





#### Process Checks
